logstash-core/src/test/java/org/logstash/ackedqueue/QueueTest.java

```diff
@@ -329,8 +329,10 @@ public void reachMaxUnread() throws IOException, InterruptedException, Execution
         Queueable element = new StringElement("foobarbaz");
         int singleElementCapacity = ByteBufferPageIO.HEADER_SIZE + ByteBufferPageIO._persistedByteCount(element.serialize().length);
 
-        Settings settings = TestSettings.volatileQueueSettings(singleElementCapacity);
-        settings.setMaxUnread(2); // 2 so we know the first write should not block and the second should
+        Settings settings = SettingsImpl.builder(
+            TestSettings.volatileQueueSettings(singleElementCapacity)
+        ).maxUnread(2) // 2 so we know the first write should not block and the second should
+        .build();
```

Comment:
@robbavey this one is the good one though right? (the one below looks messed up to me)